### PR TITLE
add per attempt errors to outbound request stats

### PR DESCRIPTION
**Pull request by ShanniLi**
r @mmihic @Raynos 

For errors per call request, only the last one is returned to the application. We should gather the error stats at the per-attempt level as well.

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -84,12 +84,25 @@ calls initiated by this service to the target service/endpoint.  MUST include
...
+standardizing operational error types is outside the scope of the core metrics.
+
+##### outbound.calls.per-attempt.operational-errors
+The total number of errors occured per attempt whilst sending outgoing
+calls initiated by this service to the target service/endpoint. These are not
+error frames but actual local socket level errors or timeout errors. SHOULD
+include an additional ``type`` tag indicating the type of error received;
+standardizing operational error types is outside the scope of the core metrics.
 
 ##### outbound.calls.app-errors
```
Comment (Raynos) [inline docs/metrics.md]: we also want per-attempt.app-errors

Comment (ShanniLi) [inline docs/metrics.md]: added.

**Comment by Raynos** [comment]
lgtm.

**Comment by mmihic** [comment]
lgtm to me too

**Comment by Raynos** [comment]
#shippingit

**Pull request merged**